tasty/src/dotty/tools/tasty/TastyFormat.scala

```diff
@@ -62,17 +62,21 @@ Standard-Section: "ASTs" TopLevelStat*
                   IMPORT         Length qual_Term Selector*                        -- import qual selectors
                   EXPORT         Length qual_Term Selector*                        -- export qual selectors
   ValOrDefDef   = VALDEF         Length NameRef type_Term rhs_Term? Modifier*      -- modifiers val name : type (= rhs)?
-                  DEFDEF         Length NameRef TypeParam* Params* returnType_Term
-                                        rhs_Term? Modifier*                        -- modifiers def name [typeparams] paramss : returnType (= rhs)?
+                  DEFDEF         Length NameRef Param* returnType_Term rhs_Term?
+                                        Modifier*                                  -- modifiers def name [typeparams] paramss : returnType (= rhs)?
   Selector      = IMPORTED              name_NameRef                               -- name, "_" for normal wildcards, "" for given wildcards
                   RENAMED               to_NameRef                                 -- => name
                   BOUNDED               type_Term                                  -- type bound
 
   TypeParam     = TYPEPARAM      Length NameRef type_Term Modifier*                -- modifiers name bounds
-  Param         = PARAM          Length NameRef type_Term rhs_Term? Modifier*      -- modifiers name : type (= rhs_Term)?. `rhsTerm` is present in the case of an aliased class parameter
-                  PARAMEND                                                         -- ends a parameter clause
+  TermParam     = PARAM          Length NameRef type_Term rhs_Term? Modifier*      -- modifiers name : type (= rhs_Term)?. `rhsTerm` is present in the case of an aliased class parameter
+                  EMPTYCLAUSE                                                      -- an empty parameter clause ()
+                  SPLITCLAUSE                                                      -- splits two non-empty parameter clauses of the same kind
```

Comment:
Here `same kind` is not really descriptive. One could think that this can be used to separate `(a: A)(b: B)` and `[A][B]` but not `(a: A)[B](c: C)` which appears internally in extensions.

Reply:
But that's exactly what it does. It separates `(a: A)(b: B)` and `[A][B]` but not `(a: A)[B](c: C)`.

Reply:
Ok. Then the awkward part would be that to encode `[A][B]` we would need a `TermParam`, though this would always be indirectly through a `Param`. For consistency, I would have expected `[A][B]` to be representable as a `TypeParam*`.

Reply:
No, TypeParam* really means a single list of type parameters. Constructors and classes will keep that restriction; it's not planned that they will allow curried type parameter lists for now.